pkg/compact/compact.go

```diff
@@ -915,40 +918,40 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 						continue
 					}
 
-					if IsIssue347Error(err) {
-						if err := RepairIssue347(workCtx, c.logger, c.bkt, c.sy.metrics.blocksMarkedForDeletion, err); err == nil {
+					if IsIssue347Error(errCompact) {
+						if errRepair := RepairIssue347(workCtx, c.logger, c.bkt, c.sy.metrics.blocksMarkedForDeletion, err); errRepair == nil {
```

Comment:
The `err` passed to `RepairIssue347()` is wrong. It should be `errCompact` (but I would suggest to reconsider just naming differently the returned param, otherwise it's too easy to introduce such bugs).

Reply:
Good catch and indeed you are correct that it would be easier to read. I will update.